GliaWidgets/Sources/ViewController/Chat/ChatViewController.swift

```diff
@@ -36,6 +38,15 @@ final class ChatViewController: EngagementViewController, PopoverPresenter {
         viewModel.event(.viewDidLoad)
     }
 
+    override func viewDidAppear(_ animated: Bool) {
```

Comment:
Could we move the logic duplicated in Chat and Call controllers into their parent EngagementViewController?
The same for ViewModels.

Reply:
No, we can't. If audio/video call is called and we have the implementation in EngagementViewController, the logic breaks because it always tries to display alert for ChatViewController, which is not in the view hierarchy. This will display warning in the log and alert is not displayed. Furthermore, If the logic is in EngagementViewController, then with Chat everything works the first time, but if I refuse the offer, and then try Chat again, alert is not shown at all.

Reply:
I will investigate a different approach, perhaps it is possible still